### PR TITLE
Fix segment fault when profiling

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: src/engine/profiler.cc [hunk 1]
```diff
@@ -68,7 +68,20 @@ OprExecStat *Profiler::AddOprStat(int dev_type, int dev_id) {
   opr_stat->dev_type = dev_type;
   opr_stat->dev_id   = dev_id;
 
-  int idx = (dev_type-1) * cpu_num_ + dev_id;
```
Comment (piiswrong) [inline src/engine/profiler.cc]: @zihaolucky

**Comment by piiswrong** [comment]
Could you explain a little more?

File: src/engine/profiler.cc (hunk 1 above)
Comment (zihaolucky) [inline src/engine/profiler.cc]: Yep?

Comment (piiswrong) [inline src/engine/profiler.cc]: Oops. I meant @ZihengJiang

**Comment by howard0su** [comment]
The original code makes the assumption on dev_type is only CPU and GPU. The corresponding value is 0 and 1. so the code use the value to calculate the offset.

Unfortunately, the assumption is wrong. CPUPinned is 3. the result of offset is wrong that caused segment fatal.

Since dev_type is a enum, I put a switch to explicit about the intention. Assume the value of enum is bad programming habit.

The crash can be repo (100%) via train ssd with 2 GPUs. I notice this crash when debug SSD example.

**Comment by piiswrong** [comment]
I see. Are you sure the the change in indexing convention won't cause incorrect parsing later?

**Comment by howard0su** [comment]
I notice one bug in the change. please hold. let me fix it.

**Comment by howard0su** [comment]
Yes. I confirmed it works as expected. and CPU pined is correctly parsed. I just loaded the result for SSD multi GPUs run to make sure.

**Pull request merged**